src/App.tsx

```diff
@@ -15,47 +15,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type IsReserved = boolean;
+
 export const App: React.FC = () => {
+  const [isReversed, setReserved] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE as SortType);
+
+  const getReorderedGoods = (
+    goods: string[],
+    oldSortType: SortType,
+    oldIsReversed: IsReserved,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((a, b) => {
+      switch (oldSortType) {
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+        default:
+          return SortType.NONE;
+      }
+    });
+
+    if (oldIsReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const sortAlphabet = () => {
```

Comment:
you can combine sortAlphabet and sortLength into one function and pass sort parameter, e.g.

```
const sortBy = (sortType) => {
  setSortType(sortType)
}
```

then you will have one function instead of two :)